mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -3256,6 +3325,114 @@ impl MmCoin for TendermintCoin {
+                Ok(Some(other)) => MmError::err(OrderCreationPreCheckError::InternalError {
+                    reason: format!(
+                        "Expected a Tendermint coin for '{}', but found '{}'.",
+                        ticker,
+                        other.ticker()
+                    ),
+                }),
+                Ok(None) => Ok(None),
+                Err(reason) => MmError::err(OrderCreationPreCheckError::PreCheckFailed { reason }),
+            }
+        }
+
+        /// Picks an HTLC coin (IRIS or NUCLEUS) based on which IBC channel is configured
+        /// and is healthy.
+        async fn get_htlc_coin(
+            coin: &TendermintCoin,
+            ctx: &MmArc,
+        ) -> Result<Option<TendermintCoin>, MmError<OrderCreationPreCheckError>> {
+            const IRIS_PREFIX: &str = "iaa";
+            const IRIS_TICKER: &str = "IRIS";
+
+            const NUCLEUS_PREFIX: &str = "nuc";
+            const NUCLEUS_TICKER: &str = "NUCLEUS";
+
+            if coin.get_healthy_ibc_channel_for_address(IRIS_PREFIX).await.is_ok() {
+                return find_tendermint_platform_coin(ctx, IRIS_TICKER).await;
+            }
+
+            if coin.get_healthy_ibc_channel_for_address(NUCLEUS_PREFIX).await.is_ok() {
+                return find_tendermint_platform_coin(ctx, NUCLEUS_TICKER).await;
+            }
+
+            MmError::err(OrderCreationPreCheckError::PreCheckFailed {
+                reason: format!("No healthy IBC channel found for {}.", coin.ticker()),
+            })
+        }
+
+        if self.wallet_only(ctx) {
+            return MmError::err(OrderCreationPreCheckError::IsWalletOnly {
+                ticker: self.ticker().to_owned(),
+            });
+        }
+
+        if rel_coin.wallet_only(ctx) {
+            return MmError::err(OrderCreationPreCheckError::IsWalletOnly {
+                ticker: rel_coin.ticker().to_owned(),
+            });
+        }
+
+        let supports_htlc = matches!(self.account_prefix.as_str(), "nuc" | "iaa");
+
+        if supports_htlc {
+            return Ok(());
+        }
+
+        // If `self` is not an HTLC-supported coin, we need to check a few things when creating the order:
+        //  - Is there an HTLC coin enabled?
+        //  - Does that HTLC network have an IBC channel configured to `self` network?
+        //  - Does that HTLC coin have enough balance to handle IBC routing?
+
+        let Some(htlc_coin) = get_htlc_coin(self, ctx).await? else {
+            return MmError::err(OrderCreationPreCheckError::PreCheckFailed {
+                reason: "No HTLC coin is currently enabled. Please enable either Iris or Nucleus.".into(),
+            });
+        };
+
+        let my_balance = htlc_coin
+            .my_balance()
+            .compat()
+            .await
+            .map_err(|e| OrderCreationPreCheckError::InternalError { reason: e.to_string() })?
+            .spendable;
+
+        let min = BigDecimal::from(2);
```

Comment:
Why 2 was chosen? should this be a const or a config option, or will this be calculated in a later stage as the minimum needed for a swap?

Reply:
> Why 2 was chosen? 

It's very good default I think, it should be enough to cover IBC and token fees all together. But ofc, we can't be %100 without testing it on different coins (which is not possible at this stage). If we see it not enough, we can increase it easily.

> hould this be a const or a config option, or will this be calculated in a later stage as the minimum needed for a swap?

To me, this is **very** internal detail to be exposed to the users.  It can't be `const` because it's `BigDecimal`. At this point all I can do is to create const for integer 2 and use it inside `BigDecimal` but I don't know how useful that will be.

Reply:
That's a good idea! But if we want to have this dynamic to make fee control easier, it makes more sense to handle this in the coins file rather than the conf file I think.

I will add a TODO.

Reply:
> it makes more sense to handle this in the coins file rather than the conf file I think

When I say config option I mean coins config option as in coins file. Will try to be more accurate next time :)